### ISSUE TITLE
Running a single unit test throws an exception but running the whole file doesn't.

### ISSUE THREAD

**Issue by QuncCccccc** [flutter/flutter#128817]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. run `flutter test <text_field_test.dart path>`, if there's no exception, then run the step 2.
2. run `flutter test --name "Dragging between multiple lines keeps the contact point at the same place on the handle on iOS" <text_field_test.dart path>`

### Expected results

The unit test "Dragging between multiple lines keeps the contact point at the same place on the handle on iOS" should pass for both command.

### Actual results

Running the single command throws exception.
![Screenshot 2023-06-13 at 11 26 52 AM](https://github.com/flutter/flutter/assets/36861262/4df5adcc-2092-4224-b689-3d684d47ddb7)
![Screenshot 2023-06-13 at 11 27 03 AM](https://github.com/flutter/flutter/assets/36861262/f4d36b8c-c4a8-4826-8ad9-163fd7b7ea93)

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[!] Flutter (Channel [user-branch], 3.12.0-4.0.pre.51, on macOS 13.4 22F66 darwin-arm64, locale en-US)
    ! Flutter version 3.12.0-4.0.pre.51 on channel [user-branch] at /Users/quncheng/Desktop/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions at https://flutter.dev/docs/get-started/install.
    • Upstream repository git@github.com:flutter/flutter.git
    • Framework revision e7d1d69544 (2 hours ago), 2023-06-13 09:34:22 -0700
    • Engine revision 12def739b1
    • Dart version 3.1.0 (build 3.1.0-199.0.dev)
    • DevTools version 2.24.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc2)
    • Android SDK at /Users/quncheng/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc2
    • Java binary at: /Applications/Android Studio with Blaze.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (version unknown)
    • Android Studio at /Applications/Android Studio with Blaze Beta.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine Android Studio version.
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.5-9884540)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio with Blaze.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.78.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available)
    • Qun的iPhone (mobile) • 00008101-000101D22641001E • ios            • iOS 16.1.1 20B101
    • macOS (desktop)     • macos                     • darwin-arm64   • macOS 13.4 22F66 darwin-arm64
    • Chrome (web)        • chrome                    • web-javascript • Google Chrome 114.0.5735.106

[✓] Network resources
    • All expected network resources are available.
```

</details>

**Comment by QuncCccccc** [comment]
Just noticed that adding a `CupertinoLocalizationsDelegate` fixes this problem, the [commit](https://github.com/flutter/flutter/pull/128725/commits/be25a7d1cd702e46b99b7f7fab80939132a2ebb3) is added in this PR: #128725. Thanks a lot for the help @justinmc 😀!

**Comment by QuncCccccc** [comment]
Close this issue because the fix (in #128725) is merged.
